rs/nervous_system/integration_tests/tests/sns_lifecycle.rs

```diff
@@ -280,18 +282,25 @@ fn test_sns_lifecycle(
             with_mainnet_nns_canister_versions,
             None,
             neurons_fund_hotkeys,
-        );
+        )
+        .await;
 
         let with_mainnet_sns_wasms = false;
-        add_wasms_to_sns_wasm(&pocket_ic, with_mainnet_sns_wasms).unwrap();
+        add_wasms_to_sns_wasm(&pocket_ic, with_mainnet_sns_wasms)
+            .await
+            .unwrap();
 
-        nns_neuron_controller_principal_ids
-            .into_iter()
-            .map(|controller_principal_id| {
-                let response = nns::governance::list_neurons(&pocket_ic, controller_principal_id);
-                (controller_principal_id, response.full_neurons)
+        stream::iter(nns_neuron_controller_principal_ids.into_iter())
```

Comment:
NCR

This could instead just be done as a for loop, right? And the "disadvantage" of that is that you start with

```
let mut result = BTreeMap::new();
```

? This isn't really bad tho.

Don't be afraid of `primitive` solutions. Be afraid of abstractions that have a "power to weight ratio" barely > 1, or worse, < 1.

Reply:
Yes, it could have been done with a primitive. I'm happy to do that actually, but I just was in the mode of keeping everything as same-ey as it was before.

Reply:
In this case it doesn't matter at all, since we should avoid nondeterminism in tests, but I think streams are actually usually a better default than for loops when both are viable, because streams operations can easily be made concurrent